### ISSUE TITLE
[Feature] Analyze new episodes on library scan

### ISSUE THREAD

**Issue by Pfuenzle** [ConfusedPolarBear/intro-skipper#22]
From what Ive seen, currently the episodes only get scanned when the task is run.
Due to this, episodes that get download only get analyzed for their intros sometime later, which:
  - Places a greater load on the server, as all series and all seasons have to be checked for changed/added episodes
  - Allows intro skipping only after running the task, which may be annoying for airing series

What do you think about analyzing the episodes as soon as they get picked up in the library scan?

I have not checked deeply into it, but I suppose newly added episodes get added to some kind of queue, as Jellyfin needs to get metadata for the episodes. 
Maybe it is possible to use this queue to analyze the episodes for their intros.

Also, there is some kind of event system (the one the Webhook plugin uses), which notifies a plugin when a new episode is added.
This event system could also be efficiently used for this (Maybe [this event?](https://jellyfin.org/docs/plugin-api/MediaBrowser.Controller.Library.ItemChangeEventArgs.html))

Using one of those two methods, analyzing episodes as soon as they are available should be possible.
I honestly only see advantages when using this method, is this something you are interested in adding to this plugin?

**Comment by Shadowghost** [comment]
That event is indeed the correct one, you can check the official trakt plugin on how to use it if you need an example.
You still need to make a differentiation between movie and episode but that souldn't be the problem

**Comment by Yankees4life** [comment]
Yeah...I would love to have this as a feature. After autoscan does it's magic, I would love to see a trigger for the file to be analyzed right after being found. This way I don't have to manually run the schedule task all the time
